### PR TITLE
generated build index, 2nd footer, cards-style display in home index

### PR DESCRIPTION
**Pull request by DanielCadiz**
## The Story
-generated build controller and index
-added columns :name and :image to tables except user and build
-generated 2nd footer for user signed in
-added cards style display in home index
-Updated relations between build and components
-added migration files build_gpus, build_rams, build_storages and build_case_fans
-deleted gpu_id, ram_id, storage_id and case_fan_id from build table
-updated schema

## Screenshots
build_index
cards_display_and_2nd_footer

## References
pcpartpicker.com

### CODE REVIEW

**Review by jarencudilla — CHANGES_REQUESTED** [review]
Try running rubocop -a via terminal on the app folder
You have failing tests.

File: db/migrate/20210715164138_add_name_and_image.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    add_column :rams, :image, :string
+    add_column :storages, :name, :string
+    add_column :storages, :image, :string
+    add_column :cases, :name, :string
+    add_column :cases, :image, :string
+    add_column :case_fans, :name, :string
+    add_column :case_fans, :image, :string
+    add_column :cpu_fans, :name, :string
+    add_column :cpu_fans, :image, :string
+  end
```
Comment (analyn-cajocson) [inline db/migrate/20210715164138_add_name_and_image.rb]: Have a separate migration file per table

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

**Comment by analyn-cajocson** [comment]
Where are the rspec models? @DanielCadiz  @jarencudilla  @AdrianIlaga  @EliezerAmbion

**Comment by jarencudilla** [comment]
> Where are the rspec models? 

We focused on the initial app for the mid-presentation later,
While @AdrianIlaga  and I worked with the scraper.

later I'll be sending a PR for table column changes and to seed our DB for the products I was able to scrape.

TLDR: After the presentation, we will start with our RSPEC.

File: db/migrate/20210715164138_add_name_and_image.rb (hunk 1 above)
Comment (jarencudilla) [reply db/migrate/20210715164138_add_name_and_image.rb]: Our thought process for this since we have multiple tables and we're just adding the same columns, we agreed to just do it in one migration file. But moving forward we will do separate migration files for each table.

Comment (DanielCadiz) [reply db/migrate/20210715164138_add_name_and_image.rb]: Good day maam, sorry for the mistake. Will be more cautious moving forward.

**Review by AdrianIlaga — CHANGES_REQUESTED** [review]
PC Parts does not belong to build, but the other way around. 
Ex:
class Case
    has_many :builds
end

Build parts has_one should be changed to belongs_to

Build parts has_many should have bridge table between build and parts

**Review by AdrianIlaga — COMMENTED** [review]
All requested changes were successfuly implemented. Ready to push

**Review by AdrianIlaga — APPROVED** [review]

**Review by jarencudilla — APPROVED** [review]
Thank you for running the tests.
